### ISSUE TITLE
Test failures on  language-server-0.15.10

### ISSUE THREAD

**Issue by polyzen** [sveltejs/language-tools#2007]
### Describe the bug

```
  1) CallHierarchyProvider
       can prepare call hierarchy:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/build/svelte-language-server/src/language-tools-language-server-0.15.10/packages/language-server/tes
t/plugins/typescript/features/CallHierarchyProvider.test.ts)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)

  2) FindComponentReferencesProvider
       finds component references:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/build/svelte-language-server/src/language-tools-language-server-0.15.10/packages/language-server/tes
t/plugins/typescript/features/FindComponentReferencesProvider.test.ts)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)

  3) TypeScript Plugin Performance Tests
       should be fast enough:
     Error: Timeout of 15000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/build/svelte-language-server/src/language-tools-language-server-0.15.10/packages/language-server/te
st/plugins/typescript/typescript-performance.test.ts)
```

`CallHierarchyProvider` passed with Node.js 20.1.0

### Reproduction

This is the build script used:
https://github.com/archlinux/svntogit-community/blob/packages/svelte-language-server/trunk/PKGBUILD

### Expected behaviour

Tests pass

### System Info

- OS: Arch Linux
- Node.js: 19.9.0/20.1.0
- language-server-0.15.10

### Which package is the issue about?

svelte-language-server

**Comment by jasonlyu123** [comment]
These tests are usually timeouts because they literally take too long. Maybe we should increase the timeout since some of it failed in our CI occasionally and might fail consistently in less performant machines. What do you think @dummdidumm?

The CallHierarchyProvider might be a new thought; it only started to fail on my machine recently. Adding a `perf` tag for that.

**Comment by dummdidumm** [comment]
I'm a bit split on increasing the timeout. If it consistently starts happening and did not before then that hints at a performance regression - though we probably rarely are the culprit, so maybe it's ok increasing it? I'm not sure ..

**Comment by jasonlyu123** [comment]
Found that Mocha has a timeout CLI flag. Does it make sense not to increase the timeout but instruct people to set a slightly higher timeout if they have trouble running the test? And for the timeout we already increased, we can rewrite them to something like `.timeout(this.timeout() * 7.5)`.

Alternatively, We can try running a dry run with a large timeout before flaky tests. The TypeScript program creation is why it needs over 2s to run. If we move it out, most of the tests afterwards won't time out that easily.
